unity-client/Assets/Tutorial/Scripts/Tutorial_ProgressBar.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections;
+using TMPro;
+using UnityEngine;
+
+namespace DCL.Tutorial
+{
+    /// <summary>
+    /// Progress bar used by several tutorial steps to indicate their progress.
+    /// </summary>
+    public class Tutorial_ProgressBar : MonoBehaviour
+    {
+        [SerializeField] Animator progressBarAnimator;
+        [SerializeField] RectTransform progressBarRectTransform;
+        [SerializeField] TMP_Text percentageText;
+        [SerializeField] float barSpeed = 3.0f;
+
+        public event Action OnNewProgressBarSizeSet;
+
+        private float maxBarWith;
+        private Coroutine progressBarCoroutine;
+
+        private void Awake()
+        {
+            maxBarWith = ((RectTransform)transform).sizeDelta.x;
+        }
+
+        private void OnDestroy()
+        {
+            if (progressBarCoroutine != null)
+            {
+                StopCoroutine(progressBarCoroutine);
+                progressBarCoroutine = null;
+            }
+        }
+
+        /// <summary>
+        /// Set a new percentage value to the progress bar. It will make the bar size change.
+        /// </summary>
+        /// <param name="percentage">A value between 0 and 100.</param>
+        /// <param name="applyAnimation">Indicates if the bar will progresively change until reach the new percentage value.</param>
+        public void SetPercentage(int percentage, bool applyAnimation = true)
+        {
+            if (percentage < 0f || percentage > 100f)
+                return;
+
+            percentageText.text = string.Format("{0}%", percentage);
+
+            if (applyAnimation)
+            {
+                progressBarAnimator.SetTrigger("SetNewPercentage");
```

Comment:
I suggest using cached hashes instead of strings by using `Animator.StringToHash`. Otherwise, this code generates the hash everytime the `SetTrigger` is called and it can be a pretty expensive operation depending on the frequency.

Reply:
I didn't know this use of the animator params. Done! Thanks!